packages/server/dataloader/gitlabLoaders.ts

```diff
@@ -0,0 +1,59 @@
+import DataLoader from 'dataloader'
+import GitLabOAuth2Manager from '../integrations/gitlab/GitLabOAuth2Manager'
+import {IGetTeamMemberIntegrationAuthQueryResult} from '../postgres/queries/generated/getTeamMemberIntegrationAuthQuery'
+import upsertTeamMemberIntegrationAuth from '../postgres/queries/upsertTeamMemberIntegrationAuth'
+import sendToSentry from '../utils/sendToSentry'
+import RootDataLoader from './RootDataLoader'
+
+export const freshGitlabAuth = (parent: RootDataLoader) => {
+  return new DataLoader<
+    {teamId: string; userId: string},
+    IGetTeamMemberIntegrationAuthQueryResult | null,
+    string
+  >(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({teamId, userId}) => {
+          const gitlabAuth = await parent.get('teamMemberIntegrationAuths').load({
+            service: 'gitlab',
+            teamId,
+            userId
+          })
+          if (!gitlabAuth) return null
+          const {expiresAt} = gitlabAuth
+          const now = new Date()
+          if (expiresAt && expiresAt < now) {
+            const {providerId, refreshToken} = gitlabAuth
+            if (!refreshToken) {
+              sendToSentry(new Error('No refresh token in gitlabAuth'), {userId})
+              return null
+            }
+            const provider = await parent.get('integrationProviders').loadNonNull(providerId)
+            const {clientId, clientSecret, serverBaseUrl} = provider
+            const manager = new GitLabOAuth2Manager(clientId!, clientSecret!, serverBaseUrl!)
+            const oauthRes = await manager.refresh(refreshToken)
+            if (oauthRes instanceof Error) return null
+            const {accessToken, refreshToken: newRefreshToken, expires_in} = oauthRes
+            const expiresAtTimestamp = new Date().getTime() + (expires_in - 30) * 1000
```

Comment:
`expires_in is in seconds. Minus 30 to be safe. *1000 to convert it to milliseconds